include/gsl/util

```diff
@@ -110,7 +111,8 @@ GSL_SUPPRESS(bounds.2) // NO-FORMAT: attribute
     // clang-format on
     constexpr T& at(T (&arr)[N], const index i)
 {
-    Expects(i >= 0 && i < narrow_cast<index>(N));
+    static_assert(N <= std::numeric_limits<std::size_t>::max() / 2, "We only support arrays up to half the bytes of the address space.");
+    Expects(i >= 0 && narrow_cast<std::size_t>(i) < N);
```

Comment:
I now reverted most of the changes. Do you still want the `static_assert` or shall the old `narrow_cast` persist?

Reply:
The original code used a `narrow_cast` so I did not change that.
When changing to `static_cast` I get a narrowing warning

> `gsl\util(116): warning C26472: Don't use a static_cast for arithmetic conversions. Use brace initialization, gsl::narrow_cast or gsl::narrow (type.1).

So that would require a `GSL_SUPPRESS(type.1)`. Are you sure you want to change to `static_cast` with `GSL_SUPPRESS` and not keep the `narrow_cast`?
If you want the `static_cast`, what about the `narrow_cast` in the other `at` overloads?

Reply:
ah that's unfortunate, it seems like the C26472 warning doesn't use the `>=0` to know that the cast is safe. In this case let's leave it as-is.